docs/auth.md

```diff
@@ -0,0 +1,66 @@
+---
+id: auth
+title: "Authorization with OAuth2"
+sidebar_label: "OAuth2"
+---
+
+**ZIO CLI** can perform interactions requiring OAuth2. OAuth2 is added to a CLI App as an `Options[OAuth2Token]`. The token can be stored and the user can specify the path, so it is not necessary to repeat authentication. We can create it as with other options:
+```scala mdoc:silent
+import zio.cli._
+import zio.cli.oauth2.OAuth2Provider
+
+val clientId = "clientId"
+
+val provider: OAuth2Provider = OAuth2Provider.Github(clientId)
+val scope: List[String] = List("repo")
+val oauth2 = Options.oauth2(provider, scope)
+```
+
+## Construction
+Currently the supported OAuth2 providers are GitHub, Google and Facebook.
```

Comment:
These are current OAuth2 providers, so maybe change the name of the section and talk about each provider separately.

Would also be nice to show the user they can make their own OAuth2 provider.